src/basic/parse-util.h

```diff
@@ -65,6 +66,11 @@ static inline int safe_atou64(const char *s, uint64_t *ret_u) {
         return safe_atollu(s, (unsigned long long*) ret_u);
 }
 
+static inline int safe_atox64(const char *s, uint64_t *ret_x) {
+        assert_cc(sizeof(uint64_t) == sizeof(unsigned long long));
+        return safe_atollx(s, (unsigned long long*) ret_x);
+}
```

Comment:
It's not a strict requirement, but it'd make review easier if the addition of helper functions like that was split out into a separate patch(es). It's easier to review, and also easier to adapt the patch series later on.